lib/sles4sap_publiccloud.pm

```diff
@@ -214,7 +214,8 @@ sub stop_hana {
         return ();
     }
     else {
-        $self->run_cmd(cmd => $cmd, runas => get_required_var("SAP_SIDADM"), timeout => $timeout);
+        my $sapadmin = lc(get_required_var('INSTANCE_SID')) . 'adm';
```

Comment:
IIRC there are other instances of `SAP_SIDADM` as well.

Reply:
I used the logic that exists in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/28be60acbb49b00d2a31a8f3239853de04e0b9dd/lib/sles4sap.pm#L184

By looking at the function's usage, I found out that `sid` is retrieved as `get_required_var('INSTANCE_SID')`, so I used that. Are there other things I need to take care of?

Reply:
Example of func call (first argument is `sid_env`): https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/28be60acbb49b00d2a31a8f3239853de04e0b9dd/lib/sles4sap.pm#L645

Reply:
Sure, but aren't there other instances of SAP_SIDADM in `lib/sles4sap_publiccloud.pm`? For example in lines 229, 309 & 321.

Reply:
Oh, now I got it. True, `get_required_var('INSTANCE_ID')` is used in a couple of other places in the file as well